formfyxer/lit_explorer.py

```diff
@@ -972,6 +972,39 @@ def get_citations(text: str, tokenized_sentences: List[str]) -> List[str]:
     return citations_with_context
 
 
+def get_sensitive_fields(fields: List[str]) -> List[str]:
+    """
+    Given a list of fields, identify those related to sensitive information. Sensitive fields include 
+    Social Security Number(SSN)), Driver's License (DL), and account numbers.
+    """
+    # NOTE: omitting CID since it has a lot of false positives.
+    field_patterns = {
+        "Social Security Number": [
+            "social[\W_]*security[\W_]*number",
+            "SSN",
+            "TIN$"
+        ],
+        "Bank Account Number": [
+            "account[\W_]*number",
+            "ABA$",
+            "routing[\W_]*number",
+            "checking"
+        ],
+        "Credit Card Number": [
+            "credit[\W_]*card",
+            "(CV[CDV]2?|CCV|CSC)"
+        ],
+        "Driver's License Number": [
+            "drivers[\W_]*license",
+            ".?DL$"
+        ]
+    }
+    text = "\n".join(fields)
+    field_regexes = {name: re.compile("|".join(patterns), re.IGNORECASE | re.MULTILINE) for name, patterns in field_patterns.items()}
+    sensitive_fields = [name for name, regex in field_regexes.items() if re.search(regex, text)]
```

Comment:
So this doesn't directly map the sensitive items to which field they are likely associated with, it's just over the whole form? Still useful, but I can imagine it being more useful if it was a `field_name`, `sensitive_type` tuple being returned. Idk if @nonprofittechy agrees.

Reply:
That would be a useful enhancement. Given our lack of specificity with the context, I can see value in not being specific too though.

If it's a simple addition, making it field specific is a good enhancement. Given that we're in the last 2 weeks of the project, I am OK with releasing the feature without that addition.

Reply:
I would prefer to link to the exact fields as well. We do the actual comparisons against the field names right now, so it seems we should be able to get the direct mapping. @BryceStevenWilley maybe you can shed some light on the problems we talked about in https://github.com/SuffolkLITLab/RateMyPDF/issues/25

tl;dr; neither the fields_old or fields maps directly to what the user can see on the form, so the advice was confusing when seen in the UI.

That might just a byproduct of the PDF forms I used for testing. Also, depending on the form, it flip-flopped between fields or fields_old as to which was the better source. To get more consistent results, we check both. I think the order of the fields is unchanged, so we can probably check both and return whichever name we prefer (I assume the new name). If there's a field name that maps more closely to the visible text, that would be even better imho.

Reply:
> tl;dr; neither the fields_old or fields maps directly to what the user can see on the form, so the advice was confusing when seen in the UI.

By this you mean that the field names aren't displayed on the PDF itself? IMO that's not the worst; we have a strong assumption in every other part of this process that you have access to a PDF editor that shows you the PDF specific name of the field (which is that `fields_old` should be).

You're right that info isn't shown in RateMyPDF at the moment. I still think this function should return per field which has sensitive info, but to avoid any confusion the RateMyPDF UI could still aggregate everything. This gives us more room in the future to maneuver, as we use FormFyxer for data analysis separate from RateMyPDF sometimes.

Reply:
Ah, that makes a lot of sense. Knowing this, I think I'll change the return type to be a dictionary of the data type with a list of field names and rename it in stats to be "sensitive data types". Form creators w/ a PDF editor would benefit from the field names and we can display the type (or perhaps both? I'll mock it up and see how it looks) in the RateMyPDF UI.